lib/tako/repository.rb

```diff
@@ -26,12 +26,12 @@ def add(shard_name, conf)
         const_set("TAKO_AR_CLASS_#{shard_name.upcase}", temporary_class)
         temporary_class.establish_connection(conf)
 
-        proxy_connections[shard_name] = temporary_class.connection
+        proxy_classes[shard_name] = temporary_class
         proxy_configs[shard_name] = conf
       end
 
       def create_proxy(shard_name)
-        Proxy.new(shard_name, proxy_connections[shard_name.to_sym])
+        Proxy.new(shard_name, proxy_classes[shard_name.to_sym].connection_without_tako)
```

Comment:
Does it makes any change? :)

Reply:
Please note that calling `ActiveRecord::ConnectionHandling#connection` connects with the database.

Currently `Tako::Repository#add` calls it. 
With these changes, `Tako::Repository#add` doesn't, but `Tako::Repository#create_proxy` does.
So querying with a new proxy makes a connection.

Reply:
Ah, I see what you mean.
Yes, we need it because the `connection` argument provided to `Tako::Proxy#initialize` can be affected by `Tako::ProxyStack.top`.

With the original code, `temporary_class.connection` in `add` method *always* returns the result of `super` in `Tako::ActiveRecordExt::ConnectionHandling#connection`  because `Tako::ProxyStack.top` is nil when loading shards.yml.

```ruby
module Tako
  module ActiveRecordExt
    module ConnectionHandling
       def connection
         return Tako::Repository.create_proxy(@force_shard).connection if force_shard?
         if Tako::ProxyStack.top
           Tako::ProxyStack.top.connection
         else
           super # Always comes here when called within `add` method
         end
       end
```

With the new code, `proxy_classes[shard_name.to_sym].connection` in `create_proxy` method can return a result of both `super` or `Tako::ProxyStack.top.connection` because `Tako::ProxyStack.top` can be non-nil. If it returns latter, shard's name and real connection conflict in a proxy object. This is same if `force_shard?` is true .

For example, without the alias, this code creates a proxy object where shard_name is `:shard02` but connection is to shard01.

```ruby
Tako.shard(:shard01) { ModelA.shard(:shard02).where(value1: 1).limit(5).count }
```

Reply:
Ahhhh!!! I got understand. In the sample case.

```rb
Tako.shard(:shard01) { ModelA.shard(:shard02).where(value1: 1).limit(5).count }
```
if `ModelA.shard(:shard02)` is first query, it might be go to `:shard01` because of nested w/o alias... 
Must be fixed. 

and there are some refactor-able magic code...!